OWL2/OWL22CASL.hs

```diff
@@ -118,7 +118,21 @@ uriToIdM = return . uriToCaslId
 
 -- | Extracts Id from URI
 uriToCaslId :: IRI -> Id
-uriToCaslId urI =  stringToId $ showIRI urI
+uriToCaslId urI = 
+ let urS = showIRI urI
+     urS' = concatMap 
+            (\c ->  if isAlpha c ||
+                       isDigit c || 
+                       c == '\'' ||
+                       c == '_'
+                   then [c]
+                   else "_u" ) urS
+ in case urS of
+      x : _ -> 
+         if isDigit x then genName urS' 
```

Comment:
this should be done if `x` is not a letter, since CASL Ids must start with a letter (aha, or a dot, see DOT-WORDS in the CASL reference manual)